lib/Service/OpenAiAPIService.php

```diff
@@ -111,68 +110,77 @@ private function isModelListValid($models): bool {
 
 	/**
 	 * @param ?string $userId
+	 * @param bool $refresh
 	 * @return array|string[]
 	 * @throws Exception
 	 */
-	public function getModels(?string $userId): array {
-		// caching against 'getModelEnumValues' calls from all the providers
-		if ($this->areCredsValid === false) {
-			$this->logger->info('Cannot get OpenAI models without an API key');
-			return [];
-		} elseif ($this->areCredsValid === null) {
-			if ($this->isUsingOpenAi() && $this->openAiSettingsService->getUserApiKey($userId, true) === '') {
-				$this->areCredsValid = false;
-				$this->logger->info('Cannot get OpenAI models without an API key');
-				return [];
+	public function getModels(?string $userId, bool $refresh = false): array {
+		$cache = $this->cacheFactory->createDistributed(Application::APP_ID);
+		if (!$refresh) {
+			if ($this->modelsMemoryCache !== null) {
+				$this->logger->debug('Getting OpenAI models from the memory cache');
+				return $this->modelsMemoryCache;
 			}
-			$this->areCredsValid = true;
-		}
 
-		if ($this->modelsMemoryCache !== null) {
-			$this->logger->debug('Getting OpenAI models from the memory cache');
-			return $this->modelsMemoryCache;
-		}
+			$userCacheKey = Application::MODELS_CACHE_KEY . '_' . ($userId ?? '');
+			$adminCacheKey = Application::MODELS_CACHE_KEY . '-main';
 
-		$userCacheKey = Application::MODELS_CACHE_KEY . '_' . ($userId ?? '');
-		$adminCacheKey = Application::MODELS_CACHE_KEY . '-main';
-		$cache = $this->cacheFactory->createDistributed(Application::APP_ID);
+			// try to get models from the user cache first
+			if ($userId !== null) {
+				$userCachedModels = $cache->get($userCacheKey);
+				if ($userCachedModels) {
+					$this->logger->debug('Getting OpenAI models from user cache for user ' . $userId);
+					return $userCachedModels;
+				}
+			}
 
-		// try to get models from the user cache first
-		if ($userId !== null) {
-			$userCachedModels = $cache->get($userCacheKey);
-			if ($userCachedModels) {
-				$this->logger->debug('Getting OpenAI models from user cache for user ' . $userId);
-				return $userCachedModels;
+			// if the user has an API key or uses basic auth, skip the admin cache
+			if (!(
+				$this->openAiSettingsService->getUserApiKey($userId, false) !== ''
+				|| (
+					$this->openAiSettingsService->getUseBasicAuth()
+					&& $this->openAiSettingsService->getUserBasicUser($userId) !== ''
+					&& $this->openAiSettingsService->getUserBasicPassword($userId) !== ''
+				)
+			)) {
+				// if no user cache or userId is null, try to get from the admin cache
```

Comment:
looks like this actually is not run if userId is null 🤔

Reply:
Jup